core/src/main/scala/org/apache/spark/util/Clock.scala

```diff
@@ -21,9 +21,36 @@ package org.apache.spark.util
  * An interface to represent clocks, so that they can be mocked out in unit tests.
  */
 private[spark] trait Clock {
-  def getTime(): Long
+  def getTimeMillis(): Long
+  def waitTillTime(targetTime: Long): Long
 }
 
-private[spark] object SystemClock extends Clock {
-  def getTime(): Long = System.currentTimeMillis()
+private[spark] class SystemClock extends Clock {
+
+  val minPollTime = 25L
+
+  def getTimeMillis(): Long = System.currentTimeMillis()
+
+  def waitTillTime(targetTime: Long): Long = {
```

Comment:
I find this is used in `org.apache.spark.streaming.util.RecurringTimer` which is still needed in `JobGenerator` and `BlockGenerator`. I can make the other change though, one sec...